jpsShared/src/org/jetbrains/jps/incremental/scala/model/HydraSettingsImpl.java

```diff
@@ -0,0 +1,50 @@
+package org.jetbrains.jps.incremental.scala.model;
+
+import com.intellij.util.xmlb.XmlSerializerUtil;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.jps.model.ex.JpsElementBase;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @author Maris Alexandru
+ */
+public class HydraSettingsImpl extends JpsElementBase<HydraSettingsImpl> implements HydraSettings{
+  public static final HydraSettings DEFAULT = new HydraSettingsImpl(new State());
+
+  private final State myState;
+
+  public HydraSettingsImpl(State state) {
+    this.myState = state;
+  }
+
+  @Override
+  public boolean isHydraEnabled() { return myState.isHydraEnabled; }
+
+  @Override
+  public String getHydraVersion() {
+    return myState.hydraVersion;
+  }
+
+  @Override
+  public Map<String, List<String>> getArtifactPaths() { return myState.artifactPaths; }
+
+  @NotNull
+  @Override
+  public HydraSettingsImpl createCopy() {
+    return new HydraSettingsImpl(XmlSerializerUtil.createCopy(myState));
+  }
+
+  @Override
+  public void applyChanges(@NotNull HydraSettingsImpl hydraSettings) {
+    //do nothing
+  }
+
+  public static class State {
+    public boolean isHydraEnabled = false;
+    public String hydraVersion = "";
+    public Map<String, List<String>> artifactPaths = new HashMap();
```

Comment:
I think we have a potential visibility problem on these fields as they are set on one thread and read by another. If that's indeed the case, let's add `@volatile` to each field declaration. By the way, I'm assuming these fields can never be accessed (read or write) in two different threads at the same time. Is this a safe assumption?